packages/core/src/manifest/ManifestYaml.tsx

```diff
@@ -1,26 +1,48 @@
 import React from 'react';
 import { Modal } from 'react-bootstrap';
+import { ArtifactService } from '../pipeline/config/triggers/artifacts/ArtifactService';
+import { decodeUnicodeBase64 } from '../utils';
 
-export interface IManifestYamlProps {
+type IManifestYamlProps = {
   linkName: string;
-  manifestText: string;
   modalTitle: string;
-}
+} & ({ manifestText: string; manifestUri?: never } | { manifestText?: never; manifestUri: string });
 
-export function ManifestYaml(props: IManifestYamlProps) {
+export function ManifestYaml({ linkName, modalTitle, manifestText, manifestUri }: IManifestYamlProps) {
   const [modalVisible, setModalVisible] = React.useState<boolean>(false);
   const toggle = () => setModalVisible(!modalVisible);
+  const [fetchedManifestText, setFetchedManifestText] = React.useState<string>('Loading...');
+  const [error, setError] = React.useState<string | null>(null);
+
+  React.useEffect(() => {
+    if (manifestUri) {
+      ArtifactService.getArtifactByContentReference(manifestUri)
```

Comment:
I'm wondering if we can memoize this call

Reply:
We could cache things since it is based on the hash of the artifact. That could be a separate PR for optimization, if necessary. I dont think these calls happpen too too often, so may be fine without it